include/openPMD/backend/PatchRecordComponent.hpp

```diff
@@ -56,14 +57,37 @@ class PatchRecordComponent : public BaseRecordComponent
     uint8_t getDimensionality() const;
     Extent getExtent() const;
 
+    /** Load a patch record
+     *
+     * In FlushType::DEFER, the data managed by the returned
+     * std::shared_ptr< T > MUST NOT be read, modified or deleted until
+     * Series::flush is called.
+     */
     template< typename T >
     std::shared_ptr< T > load();
+
+    /** Load a patch record into pre-allocated memory
+     *
+     * In FlushType::DEFER, the data managed by the passed
+     * std::shared_ptr< T > MUST NOT be read, modified or deleted until
+     * Series::flush is called.
+     */
     template< typename T >
     void load(std::shared_ptr< T >);
+
+    /** Store a patch record
+     *
+     * In FlushType::DEFER, the passed data MUST NOT be modified or deleted
+     * until Series::flush is called.
+     */
     template< typename T >
     void store(uint64_t idx, T);
+
     //! @todo add support for constant patch record components
 
+    //! @todo allow to read all patch records at once in a convienient object
+    //!       to iterate over
```

Comment:
There's an issue for this already (https://github.com/openPMD/openPMD-api/issues/361), maybe mention it here.